console/src/main/java/org/nervos/ckb/Wallet.java

```diff
@@ -127,11 +141,26 @@ private CellInputs getCellInputs(String lockHash, BigInteger needCapacities) thr
     return new CellInputs(cellInputs, new BigDecimal(inputsCapacities).toBigInteger());
   }
 
-  class CellInputs {
+  private Script generateLockScript(String privateKey, String codeHash) {
+    String publicKey = Sign.publicKeyFromPrivate(Numeric.toBigInt(privateKey), true).toString(16);
+    String blake160 =
+        Numeric.prependHexPrefix(Numeric.cleanHexPrefix(Hash.blake2b(publicKey)).substring(0, 40));
+    return new Script(codeHash, Collections.singletonList(blake160), Script.TYPE);
```

Comment:
can I generate a script which hash_type is data?

Reply:
Yes. Last parameter is hash_type, anyone could set value.